Fix shellcheck error

it is causing all PRs to be marked as red which makes it hard to know if the PR is correct or not.

```
#!/bin/sh -eo pipefail
find . -type f -name '*.sh' | wc -l
find . -type f -name '*.sh' | xargs shellcheck --external-sources
12

In ./check_plugins_location_v2.sh line 19:
declare -a arr=(`find "$1" -name "meta.yaml"`)
                ^---------------------------^ SC2207: Prefer mapfile or read -a to split command output (or quote to avoid splitting).
                ^---------------------------^ SC2006: Use $(...) notation instead of legacy backticked `...`.


In ./check_plugins_location_v2.sh line 22:
    plugin_id=$(evaluate_plugin_id $i)
                                   ^-- SC2086: Double quote to prevent globbing and word splitting.


In ./check_plugins_location_v1.sh line 15:
declare -a arr=(`find plugins -name "meta.yaml"`)
                ^-- SC2207: Prefer mapfile or read -a to split command output (or quote to avoid splitting).
                ^-- SC2006: Use $(...) notation instead of legacy backticked `...`.


In ./set_plugin_dates.sh line 17:
  if [ $? -ne 0 ]; then
       ^-- SC2181: Check exit code directly with e.g. 'if mycmd;', not indirectly with $?.


In ./set_plugin_dates.sh line 27:
declare -a arr=(`find . -name "meta.yaml"`)
                ^------------------------^ SC2207: Prefer mapfile or read -a to split command output (or quote to avoid splitting).
                ^------------------------^ SC2006: Use $(...) notation instead of legacy backticked `...`.


In ./check_plugins_viewer_mandatory_fields_v3.sh line 47:
declare -a arr=(`find v3 -name "meta.yaml"`)
                ^-------------------------^ SC2207: Prefer mapfile or read -a to split command output (or quote to avoid splitting).
                ^-------------------------^ SC2006: Use $(...) notation instead of legacy backticked `...`.


In ./check_plugins_viewer_mandatory_fields_v3.sh line 50:
    plugin_id=$(evaluate_plugin_id $i)
                                   ^-- SC2086: Double quote to prevent globbing and word splitting.


In ./check_plugins_viewer_mandatory_fields_v3.sh line 58:
      VALUE=$(yq r $i "$FIELD")
                   ^-- SC2086: Double quote to prevent globbing and word splitting.


In ./check_plugins_viewer_mandatory_fields_v1.sh line 45:
declare -a arr=(`find plugins -name "meta.yaml"`)
                ^-- SC2207: Prefer mapfile or read -a to split command output (or quote to avoid splitting).
                ^-- SC2006: Use $(...) notation instead of legacy backticked `...`.


In ./check_plugins_viewer_mandatory_fields_v1.sh line 58:
      VALUE=$(yq r $i "$FIELD")
                   ^-- SC2086: Double quote to prevent globbing and word splitting.


In ./cico_build.sh line 53:
  docker tag ${IMAGE} $TARGET
             ^------^ SC2086: Double quote to prevent globbing and word splitting.
                      ^-----^ SC2086: Double quote to prevent globbing and word splitting.


In ./cico_build.sh line 54:
  docker push $TARGET
              ^-----^ SC2086: Double quote to prevent globbing and word splitting.


In ./cico_build.sh line 61:
  if [ $TARGET == "rhel" ]; then
       ^-----^ SC2086: Double quote to prevent globbing and word splitting.


In ./cico_build.sh line 69:
  if [ -n "${QUAY_USERNAME}" -a -n "${QUAY_PASSWORD}" ]; then
                             ^-- SC2166: Prefer [ p ] && [ q ] as [ p -a q ] is not well defined.


In ./cico_build.sh line 70:
    docker login -u ${QUAY_USERNAME} -p ${QUAY_PASSWORD} ${REGISTRY}
                    ^--------------^ SC2086: Double quote to prevent globbing and word splitting.
                                        ^--------------^ SC2086: Double quote to prevent globbing and word splitting.


In ./cico_build.sh line 78:
  TAG=$(echo $GIT_COMMIT | cut -c1-${DEVSHIFT_TAG_LEN})
             ^---------^ SC2086: Double quote to prevent globbing and word splitting.
                                   ^-----------------^ SC2086: Double quote to prevent globbing and word splitting.


In ./cico_build.sh line 80:
  tag_push ${REGISTRY}/openshiftio/$IMAGE:$TAG
                                          ^--^ SC2086: Double quote to prevent globbing and word splitting.


In ./check_plugins_images.sh line 16:
declare -a arr=(`find . -name "meta.yaml"`)
                ^------------------------^ SC2207: Prefer mapfile or read -a to split command output (or quote to avoid splitting).
                ^------------------------^ SC2006: Use $(...) notation instead of legacy backticked `...`.


In ./check_plugins_images.sh line 19:
    ICON=$(yq r $i icon | sed 's/^"\(.*\)"$/\1/')
                ^-- SC2086: Double quote to prevent globbing and word splitting.


In ./check_plugins_images.sh line 22:
      plugin_id=$(evaluate_plugin_id $i)
                                     ^-- SC2086: Double quote to prevent globbing and word splitting.


In ./check_plugins_viewer_mandatory_fields_v2.sh line 45:
declare -a arr=(`find v2 -name "meta.yaml"`)
                ^-------------------------^ SC2207: Prefer mapfile or read -a to split command output (or quote to avoid splitting).
                ^-------------------------^ SC2006: Use $(...) notation instead of legacy backticked `...`.


In ./check_plugins_viewer_mandatory_fields_v2.sh line 48:
    plugin_id=$(evaluate_plugin_id $i)
                                   ^-- SC2086: Double quote to prevent globbing and word splitting.


In ./check_plugins_viewer_mandatory_fields_v2.sh line 56:
      VALUE=$(yq r $i "$FIELD")
                   ^-- SC2086: Double quote to prevent globbing and word splitting.


In ./index.sh line 29:
    declare -a arr=(`find "$1" -name "meta.yaml"`)
                    ^---------------------------^ SC2207: Prefer mapfile or read -a to split command output (or quote to avoid splitting).
                    ^---------------------------^ SC2006: Use $(...) notation instead of legacy backticked `...`.


In ./index.sh line 42:
        plugin_id=$(getId $i)
                          ^-- SC2086: Double quote to prevent globbing and word splitting.


In ./index.sh line 45:
        for field in ${fields[@]}
                     ^----------^ SC2068: Double quote array expansions to avoid re-splitting elements.


In ./index.sh line 47:
            echo "  \"$field\":\""$(yq r "$i" "$field" | sed 's/^"\(.*\)"$/\1/')"\","
                                  ^-- SC2027: The surrounding quotes actually unquote this. Remove or escape them.
                                  ^-- SC2046: Quote this to prevent word splitting.


In ./index.sh line 63:
        echo "  \"links\": {\"self\":\"/$(echo $i)\" }"
                                        ^--------^ SC2116: Useless echo? Instead of 'cmd $(echo foo)', just use 'cmd foo'.
                                               ^-- SC2086: Double quote to prevent globbing and word splitting.


In ./index_v2.sh line 23:
    evaluate_plugin_id $1
                       ^-- SC2086: Double quote to prevent globbing and word splitting.


In ./index_v2.sh line 32:
    declare -a arr=(`find "$1" -name "meta.yaml"`)
                    ^---------------------------^ SC2207: Prefer mapfile or read -a to split command output (or quote to avoid splitting).
                    ^---------------------------^ SC2006: Use $(...) notation instead of legacy backticked `...`.


In ./index_v2.sh line 45:
        plugin_id=$(getId $i)
                          ^-- SC2086: Double quote to prevent globbing and word splitting.


In ./index_v2.sh line 48:
        for field in ${fields[@]}
                     ^----------^ SC2068: Double quote array expansions to avoid re-splitting elements.


In ./index_v2.sh line 50:
            echo "  \"$field\":\""$(yq r "$i" "$field" | sed 's/^"\(.*\)"$/\1/')"\","
                                  ^-- SC2027: The surrounding quotes actually unquote this. Remove or escape them.
                                  ^-- SC2046: Quote this to prevent word splitting.


In ./index_v2.sh line 66:
        echo "  \"links\": {\"self\":\"/$(echo $i|sed 's/\/meta.yaml$//g')\" }"
                                               ^-- SC2086: Double quote to prevent globbing and word splitting.

For more information:
  https://www.shellcheck.net/wiki/SC2068 -- Double quote array expansions to ...
  https://www.shellcheck.net/wiki/SC2027 -- The surrounding quotes actually u...
  https://www.shellcheck.net/wiki/SC2046 -- Quote this to prevent word splitt...
Exited with code 123
```

--- a/check_plugins_images.sh
+++ b/check_plugins_images.sh
@@ -13,13 +13,13 @@ set -e
 source ./util.sh
 
 ## check that icon tags in meta.yaml files points to the .svg images
-declare -a arr=(`find . -name "meta.yaml"`)
+readarray -d '' arr < <(find . -name 'meta.yaml' -print0)
 for i in "${arr[@]}"
 do
-    ICON=$(yq r $i icon | sed 's/^"\(.*\)"$/\1/')
+    ICON=$(yq r "$i" icon | sed 's/^"\(.*\)"$/\1/')
     # Regex: contains .svg and not contains dots after it (to avoid xxx.svg.jpg hacks)
     if [[ ! $ICON =~ (\.svg)+[^\.]*$ ]]; then
-      plugin_id=$(evaluate_plugin_id $i)
+      plugin_id=$(evaluate_plugin_id "$i")
       plugin_version=$(yq r "$i" version | sed 's/^"\(.*\)"$/\1/')
       plugin_publisher=$(yq r "$i" publisher | sed 's/^"\(.*\)"$/\1/')
       echo "!!!   Wrong icon type found in '${plugin_id}' of publisher '${plugin_publisher}' with version '${plugin_version}':"

--- a/check_plugins_location_v1.sh
+++ b/check_plugins_location_v1.sh
@@ -12,7 +12,7 @@ set -e
 
 source ./util.sh
 
-declare -a arr=(`find plugins -name "meta.yaml"`)
+readarray -d '' arr < <(find plugins -name 'meta.yaml' -print0)
 for i in "${arr[@]}"
 do
     id=$(yq r "$i" id | sed 's/^"\(.*\)"$/\1/')

--- a/check_plugins_location_v2.sh
+++ b/check_plugins_location_v2.sh
@@ -16,10 +16,10 @@ set -e
 
 source ./util.sh
 
-declare -a arr=(`find "$1" -name "meta.yaml"`)
+readarray -d '' arr < <(find "$1" -name 'meta.yaml' -print0)
 for i in "${arr[@]}"
 do
-    plugin_id=$(evaluate_plugin_id $i)
+    plugin_id=$(evaluate_plugin_id "$i")
 
     expected_path="$1/plugins/${plugin_id}/meta.yaml"
     if [[ "${expected_path}" != "$i" ]]; then

--- a/check_plugins_viewer_mandatory_fields_v1.sh
+++ b/check_plugins_viewer_mandatory_fields_v1.sh
@@ -42,7 +42,7 @@ function check_category() {
   return 1
 }
 
-declare -a arr=(`find plugins -name "meta.yaml"`)
+readarray -d '' arr < <(find plugins -name 'meta.yaml' -print0)
 for i in "${arr[@]}"
 do
     id=$(yq r "$i" id | sed 's/^"\(.*\)"$/\1/')
@@ -55,7 +55,7 @@ do
 
     for FIELD in "${FIELDS[@]}"
     do
-      VALUE=$(yq r $i "$FIELD")
+      VALUE=$(yq r "$i" "$FIELD")
       if [[ "${FIELD}" == "category" ]];then
         if ! check_category "$i" "${VALUE}";then
           echo "!!!   Invalid category in '${full_id}': $VALUE"

--- a/check_plugins_viewer_mandatory_fields_v2.sh
+++ b/check_plugins_viewer_mandatory_fields_v2.sh
@@ -42,10 +42,10 @@ function check_category() {
   return 1
 }
 
-declare -a arr=(`find v2 -name "meta.yaml"`)
+readarray -d '' arr < <(find v2 -name 'meta.yaml' -print0)
 for i in "${arr[@]}"
 do
-    plugin_id=$(evaluate_plugin_id $i)
+    plugin_id=$(evaluate_plugin_id "$i")
 
     echo "Checking plugin '${plugin_id}'"
 
@@ -53,7 +53,7 @@ do
 
     for FIELD in "${FIELDS[@]}"
     do
-      VALUE=$(yq r $i "$FIELD")
+      VALUE=$(yq r "$i" "$FIELD")
       if [[ "${FIELD}" == "category" ]];then
         if ! check_category "$i" "${VALUE}";then
           echo "!!!   Invalid category in '${plugin_id}': $VALUE"

--- a/check_plugins_viewer_mandatory_fields_v3.sh
+++ b/check_plugins_viewer_mandatory_fields_v3.sh
@@ -44,10 +44,10 @@ function check_category() {
   return 1
 }
 
-declare -a arr=(`find v3 -name "meta.yaml"`)
+readarray -d '' arr < <(find v3 -name 'meta.yaml' -print0)
 for i in "${arr[@]}"
 do
-    plugin_id=$(evaluate_plugin_id $i)
+    plugin_id=$(evaluate_plugin_id "$i")
 
     echo "Checking plugin '${plugin_id}'"
 
@@ -55,7 +55,7 @@ do
 
     for FIELD in "${FIELDS[@]}"
     do
-      VALUE=$(yq r $i "$FIELD")
+      VALUE=$(yq r "$i" "$FIELD")
       if [[ "${FIELD}" == "category" ]];then
         if ! check_category "$i" "${VALUE}";then
           echo "!!!   Invalid category in '${plugin_id}': $VALUE"

--- a/cico_build.sh
+++ b/cico_build.sh
@@ -50,15 +50,15 @@ function install_deps() {
 
 function tag_push() {
   local TARGET=$1
-  docker tag ${IMAGE} $TARGET
-  docker push $TARGET
+  docker tag "${IMAGE}" "$TARGET"
+  docker push "$TARGET"
 }
 
 function deploy() {
   TARGET=${TARGET:-"centos"}
   REGISTRY="quay.io"
 
-  if [ $TARGET == "rhel" ]; then
+  if [ "$TARGET" == "rhel" ]; then
     DOCKERFILE="Dockerfile.rhel"
     IMAGE="rhel-che-plugin-registry"
   else
@@ -66,8 +66,8 @@ function deploy() {
     IMAGE="che-plugin-registry"
   fi
 
-  if [ -n "${QUAY_USERNAME}" -a -n "${QUAY_PASSWORD}" ]; then
-    docker login -u ${QUAY_USERNAME} -p ${QUAY_PASSWORD} ${REGISTRY}
+  if [ -n "${QUAY_USERNAME}" ] && [ -n "${QUAY_PASSWORD}" ]; then
+    docker login -u "${QUAY_USERNAME}" -p "${QUAY_PASSWORD}" "${REGISTRY}"
   else
     echo "Could not login, missing credentials for the registry"
   fi
@@ -75,10 +75,10 @@ function deploy() {
   # Let's deploy
   docker build -t ${IMAGE} -f ${DOCKERFILE} .
 
-  TAG=$(echo $GIT_COMMIT | cut -c1-${DEVSHIFT_TAG_LEN})
+  TAG=$(echo "$GIT_COMMIT" | cut -c1-"${DEVSHIFT_TAG_LEN}")
 
-  tag_push ${REGISTRY}/openshiftio/$IMAGE:$TAG
-  tag_push ${REGISTRY}/openshiftio/$IMAGE:latest
+  tag_push "${REGISTRY}/openshiftio/$IMAGE:$TAG"
+  tag_push "${REGISTRY}/openshiftio/$IMAGE:latest"
   echo 'CICO: Image pushed, ready to update deployed app'
 }
 

--- a/index.sh
+++ b/index.sh
@@ -26,7 +26,7 @@ function getId() {
 function buildIndex() {
     fields=('displayName' 'version' 'type' 'name' 'description' 'publisher')
     ## search for all editors and plugins
-    declare -a arr=(`find "$1" -name "meta.yaml"`)
+    readarray -d '' arr < <(find "$1" -name 'meta.yaml' -print0)
     FIRST_LINE=true
     echo "["
     ## now loop through meta files
@@ -39,12 +39,13 @@ function buildIndex() {
             echo ",{"
         fi
 
-        plugin_id=$(getId $i)
+        plugin_id=$(getId "$i")
         echo "  \"id\": \"$plugin_id\","
 
-        for field in ${fields[@]}
+        for field in "${fields[@]}"
         do
-            echo "  \"$field\":\""$(yq r "$i" "$field" | sed 's/^"\(.*\)"$/\1/')"\","
+            value="$(yq r "$i" "$field" | sed 's/^"\(.*\)"$/\1/')"
+            echo "  \"$field\":\"$value\","
         done
 
         # Add deprecate section
@@ -60,7 +61,7 @@ function buildIndex() {
             echo "  },"
         fi
 
-        echo "  \"links\": {\"self\":\"/$(echo $i)\" }"
+        echo "  \"links\": {\"self\":\"/$i\" }"
         echo "}"
     done
     echo "]"

--- a/index_v2.sh
+++ b/index_v2.sh
@@ -20,7 +20,7 @@ source ./util.sh
 # Arguments:
 # 1 - meta.yaml location
 function getId() {
-    evaluate_plugin_id $1
+    evaluate_plugin_id "$1"
 }
 
 # getId function MUST be defined to use this function
@@ -29,7 +29,7 @@ function getId() {
 function buildIndex() {
     fields=('displayName' 'version' 'type' 'name' 'description' 'publisher')
     ## search for all editors and plugins
-    declare -a arr=(`find "$1" -name "meta.yaml"`)
+    readarray -d '' arr < <(find "$1" -name 'meta.yaml' -print0)
     FIRST_LINE=true
     echo "["
     ## now loop through meta files
@@ -42,12 +42,13 @@ function buildIndex() {
             echo ",{"
         fi
 
-        plugin_id=$(getId $i)
+        plugin_id=$(getId "$i")
         echo "  \"id\": \"$plugin_id\","
 
-        for field in ${fields[@]}
+        for field in "${fields[@]}"
         do
-            echo "  \"$field\":\""$(yq r "$i" "$field" | sed 's/^"\(.*\)"$/\1/')"\","
+            value="$(yq r "$i" "$field" | sed 's/^"\(.*\)"$/\1/')"
+            echo "  \"$field\":\"$value\","
         done
 
         # Add deprecate section
@@ -63,7 +64,8 @@ function buildIndex() {
             echo "  },"
         fi
 
-        echo "  \"links\": {\"self\":\"/$(echo $i|sed 's/\/meta.yaml$//g')\" }"
+        path=$(echo "$i" | sed 's/\/meta.yaml$//g')
+        echo "  \"links\": {\"self\":\"/$path\" }"
         echo "}"
     done
     echo "]"

--- a/set_plugin_dates.sh
+++ b/set_plugin_dates.sh
@@ -13,8 +13,7 @@ set -e
 is_first_publication_date_present() {
   # check that first publication date is present in yaml,
   # and is not an null or empty value
-  VALUE=$(yq r "$1" firstPublicationDate)
-  if [ $? -ne 0 ]; then
+  if ! VALUE=$(yq r "$1" firstPublicationDate); then
     exit 1
   fi
 
@@ -24,7 +23,7 @@ is_first_publication_date_present() {
   return 0;
 }
 
-declare -a arr=(`find . -name "meta.yaml"`)
+readarray -d '' arr < <(find . -name 'meta.yaml' -print0)
 for i in "${arr[@]}"
 do
     DATE=$(date -I)

